launcher/icons/IconList.cpp

```diff
@@ -425,3 +494,15 @@ QString IconList::getDirectory() const
 {
     return m_dir.absolutePath();
 }
+
+/// Returns the directory of the icon with the given key or the default directory if it's a builtin icon.
+QString IconList::iconDirectory(const QString& key) const
+{
+    for (const auto& mmc_icon : icons) {
+        if (mmc_icon.m_key == key && mmc_icon.has(IconType::FileBased)) {
+            QFileInfo icon_file(mmc_icon.getFilePath());
+            return icon_file.dir().path();
```

Comment:
Why not just return the absolute path of the icon(https://doc.qt.io/qt-6/qfileinfo.html#absolutePath)
```suggestion
            return icon_file.absolutePath();
```